tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +863,49 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCanBeIteratedMultipleTimes()
+    {
+        $skippedMessage = <<<'MSG'
+Iterators are not rewindable for either PDO or Non PDO drivers. This is because PDO by default is not rewindable and 
+to be consistent we have not implemented rewindable iterators for non PDO drivers. Should PDO ever become rewindable by
+default we should wrap the legacy drivers in Doctrine\DBAL\RewindableGenerator for example on the Mysqli driver you 
+would do something like the following:
+
+return new RewindableGenerator(function() {
+    $this->_stmt->data_seek(0);
+    while ($row = $this->fetch()) {
+        yield $row;
+    }
+});
+
+If this has been done you can include this test. 
+MSG;
+
+        $this->markTestSkipped($skippedMessage);
```

Comment:
Thanks for the additional info, have pushed a commit which I *think* does what you are talking about, would you mind taking a look and letting me know? https://github.com/doctrine/dbal/pull/2718/commits/07d5f6f8bf5fff49eb1f4f4efd5bf3e59ad377d0